### PR TITLE
Add syntax and semantic tests for underscore

### PR DESCRIPTION
**Pull request by axic**
Part of #9604.

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `35f69baaae793c1e69eb915fdf38c4cc02f8cd84`:
```
Error: Trailing whitespace found:
 test/libsolidity/semanticTests/underscore/in_modifier.sol:11: 
 test/libsolidity/syntaxTests/underscore/as_function.sol:4: 
 test/libsolidity/syntaxTests/underscore/in_modifier.sol:11: 

```
Please check that your changes are working as intended.

File: test/libsolidity/semanticTests/underscore/as_function.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        _;
+        return 33;
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// _() -> 88
+// g() -> 88
+// h() -> 33
```
Comment (hrkrshnn) [inline test/libsolidity/semanticTests/underscore/as_function.sol:20]: If these instances are going to be disallowed, does it make sense to have them as semantic tests?

Comment (axic) [reply test/libsolidity/semanticTests/underscore/as_function.sol:20]: We'll see what happens when they are disallowed, right? I personally feel it is reassuring to see things break when I expect them to break and to update tests then and there.

Comment (hrkrshnn) [reply test/libsolidity/semanticTests/underscore/as_function.sol:20]: Ok! The syntax tests are fine though.

**Review by hrkrshnn — DISMISSED** [review]

**Comment by chriseth** [comment]
tests are failing

**Comment by axic** [comment]
The tests result were generated on macos using isoltest. I did not manually type them up.

So no idea why it would do this:
````
ASSERTION FAILURE:
- file   : boostTest.cpp
- line   : 114
- message: Test expectation mismatch.
Expected result:
// f() -> 88
// g() -> FAILUREhex"08c379a0", 0x20, 6, "failed"

Obtained result:
// f() -> 88
// g() -> FAILURE
Warning: The call to "g()" returned 


Attention: Updates on the test will apply the detected format displayed.
```

**Comment by axic** [comment]
Ah it is the goddamn evm version. Why do we have to make the testing suite so complicated and developer unfriendly?

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**